### PR TITLE
Movielens setup

### PR DESCRIPTION
**Pull request by naveenram00**
This branch contains the code which build the tsvs for training the ml_tags and ml_sequences tasks. You can start reviewing it now but I will probably add a few more tests in the next day or two.

### CODE REVIEW

File: data/build_movielens.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+flags.DEFINE_integer("sample_rate", 3, \
+  "rate for sampling multiple masked examples from one ml_tags example")
+FLAGS = flags.FLAGS
+
+def main(_):
+  """Builds the movielens datasets and saves tsvs in output_dir"""
+  # Define filepaths
+  paths = {
+      "sequences": sorted(glob.glob(os.path.join(FLAGS.movielens_dir,
+                                                 "sequences", "*.csv*"))),
```
Comment (log0) [inline data/build_movielens.py:45]: what's the extra * at the end of "*.csv"?

File: data/build_movielens.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+      "movies": glob.glob(os.path.join(FLAGS.movielens_dir,
+                                       "ml-25m/movies.csv"))[0],
+      "tags": glob.glob(os.path.join(FLAGS.movielens_dir,
+                                     "ml-25m/tags.csv"))[0],
+      "genome_tags": glob.glob(os.path.join(FLAGS.movielens_dir,
+                                            "ml-25m/genome-tags.csv"))[0],
+      "genome_scores": glob.glob(os.path.join(FLAGS.movielens_dir,
```
Comment (log0) [inline data/build_movielens.py:52]: if you are just pointing to a file, do you need to glob? same for L47, 49, 51.

File: data/build_movielens.py [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+  # Load the movie and tag id files
+  genome_tags = pd.read_csv(paths["genome_tags"])
+  tag_decoder = dict(zip(genome_tags.tagId, genome_tags.tag))
+  movies = pd.read_csv(paths["movies"])
+  movies["title"] = movies["title"].apply(flip_titles)
+  movie_decoder = dict(zip(movies.movieId, movies.title))
+  genre_decoder = dict(zip(movies.title, movies.genres))
+
+  if FLAGS.task in ["ml_sequences", "all"]:
```
Comment (log0) [inline data/build_movielens.py:64]: Suggest to wrap in a function and explicitly call out the arguments, as this if is growing a bit big. Same for the other if below.

File: data/build_movielens.py [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+      """map a list of movie names to a tab-separated string
+      example:
+      """
+      inputs = "@ %s @" % " @ ".join(arr[:-1])
+      targets = arr[-1]
+      return "\t".join((inputs, targets))
+
+
+    seqs_formatted = list(map(format_sequence, user_seqs))
+    seqs_train, seqs_test = train_test_split(seqs_formatted, test_size=.2,
```
Comment (log0) [inline data/build_movielens.py:83]: Make test_size flag?

File: data/build_movielens.py [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+
+    def format_tags(example):
+      """format one example of tags into tab-separated string"""
+      movie, tags = example
+      return "\t".join((movie, ", ".join(tags)))
+    tags_formatted = list(map(format_tags, tags_dict.items()))
+
+    if FLAGS.mask:
+      tags_formatted = list(flat_map(mask_multple, tqdm.tqdm(tags_formatted)))
+    tags_train, tags_test = train_test_split(tags_formatted, test_size=.2,
```
Comment (log0) [inline data/build_movielens.py:118]: ditto make flag

File: data/build_movielens.py (hunk 1 above)
Comment (naveenram00) [reply data/build_movielens.py:45]: the apache beam output files look like user_watch_seq.csv-00004-of-00005 so the extra * is for the numbers after csv

File: data/build_movielens.py (hunk 2 above)
Comment (naveenram00) [reply data/build_movielens.py:52]: yeah I don't know why I did that. Fixed.

File: data/build_movielens.py [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+    result.append(mask_text(ex))
+  return result
+
+def mask_text(ex):
+  """masks using the strategy described in the t5 paper section 3.1.4
+  (https://arxiv.org/abs/1910.10683) 15% of tokens are replaced with sentinel
+  mask tokens"""
+  movie, tags = ex.split("\t")
+  tokens = np.append(movie, tags.split(", "))
+  indecies = np.random.choice(len(tokens), int(np.ceil(len(tokens)*.15)))
```
Comment (log0) [inline data/build_movielens.py:153]: is 15% always fixed? maybe make this flag too.

**Pull request merged**